src/lambdaconnect_model/scoping.clj

```diff
@@ -328,29 +310,125 @@
+  "Takes a snapshot, a user object from DB, entities-by-name, parsed (and validated) EDN of scoping rules, a set of desired tags and push?.
+  A typical invocation looks like this: 
+  (scope-selected-tags config (d/db db/conn) user entities-by-name validated-scope desired-tags #{:RARestaurant.ofOwner})))
+  Returns a map with db ids, something like:
+  {:RAOwner.me #{11122, 1222} :user #{2312312}}
+  "
+  [config snapshot user entities-by-name scoping-defintion tags push?]
+  (let [relevant-rules (->> scoping-defintion
+                            (filter (fn [[_ description]]
+                                      (and (:constraint description)
+                                           (or (not push?)
+                                               (-> description :permissions :create)
+                                               (-> description :permissions :modify)
+                                               (-> description :permissions :include-in-push)))))
+                            (map (fn [[tag description]] [tag (:constraint description)]))
+                            (into {}))
+        queries (scoping-step
+                 snapshot
+                 #{(:db/id user)}
+                 entities-by-name
+                 {}
+                 #{:user}
+                 relevant-rules
+                 {:user {:dependencies #{} :rules []}})
+        filtered-queries (into {} (filter (fn [[tag _]] (contains? tags tag)) queries))]
+    (->> filtered-queries
+         (pmap (fn [x] (apply (partial execute-query config) x)))
+         (into {}))))
+
+(defn get-scoping-queries
+  "Takes entities-by-name, parsed (and validated) EDN of scoping rules, a set of desired tags and push?.
+   A typical invocation looks like this: 
+   (get-scoping-queries entities-by-name validated-scope desired-tags #{:RARestaurant.ofEmployee :RARestaurant.ofOwner})))
+   Returns a map with tags and queries generated to obtain entities for indicated user:
+  {:RAEmployee.ofOwner
+  [:find
+  ?RAEmployee-ofOwner
+  :in
+  $
+  [?user ...]
+  :where
+  [?user :app/uuid ?G__33023]
+  [(= ?G__33024 ?G__33023)]
+  [?RAOwner-me :RAOwner/internalUserId ?G__33024]
+  [?RAEmployee-ofOwner :RAEmployee/owner ?RAOwner-me]]}
+   ...
+  "
+  [entities-by-name scoping-defintion tags push?]
+  (let [user {:db/id 1111111}
+        snapshot {}
+        relevant-rules (->> scoping-defintion
+                            (filter (fn [[_ description]]
+                                      (and (:constraint description)
+                                           (or (not push?)
+                                               (-> description :permissions :create)
+                                               (-> description :permissions :modify)
+                                               (-> description :permissions :include-in-push)))))
+                            (map (fn [[tag description]] [tag (:constraint description)]))
+                            (into {}))
```

Comment:
kawałek kodu
```
(->> scoping-defintion
                             (filter (fn [[_ description]]
                                       (and (:constraint description)
                                            (or (not push?)
                                                (-> description :permissions :create)
                                                (-> description :permissions :modify)
                                                (-> description :permissions :include-in-push)))))
                             (map (fn [[tag description]] [tag (:constraint description)]))
                             (into {}))
```
Występuje chyba w 3 kolejnych funkcjach. Może jakaś funkcja?

Reply:
Myślałem, że funkcja, która wykonuje zapytania będzie korzystać z funkcji, która zwraca same zapytania.

Reply:
to w sumie miałby sens...